app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -584,25 +590,74 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
         positionIsValid = shouldConsiderPoseAsValid(mHandJoints[i].pose);
 #endif
         if (positionIsValid) {
-            float radius = mHandJoints[i].radius;
-            vrb::Matrix scale = vrb::Matrix::Identity().ScaleInPlace(vrb::Vector(radius, radius, radius));
-            transform.PostMultiplyInPlace(scale);
-
             if (renderMode == device::RenderMode::StandAlone)
                 transform.TranslateInPlace(kAverageHeight);
         } else {
             // This effectively hides the joint.
             transform.ScaleInPlace(vrb::Vector(0.0f, 0.0f, 0.0f));
         }
 
-        memcpy(&jointTransforms[i], &transform, sizeof(vrb::Matrix));
+        jointTransforms[i] = transform;
+    }
+
+    // Check whether palm of left hand is facing the head, to show navigate-back button
+    bool leftPalmFacesHead = false;
+    if (mHandeness == OpenXRHandFlags::Left) {
+#if defined(OCULUSVR)
+        if (mHandJoints[XR_HAND_JOINT_PALM_EXT].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT)
+            leftPalmFacesHead = !mHasAimState;
+#else
+        if (mHandJoints[XR_HAND_JOINT_WRIST_EXT].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
+            vrb::Vector vector = vrb::Vector(0.0f, 0.0f, 1.0f);
+            vrb::Matrix palmMatrix = jointTransforms[XR_HAND_JOINT_WRIST_EXT];
```

Comment:
Why are you using the wrist joint instead of the palm joint? Seems weird because all the variables are related to palm.